docs/manual/en/manual.txt

```diff
@@ -0,0 +1,256 @@
+
+- For 2x2op algorithms, each voice occupies two channels, assigned in the same way as for 2op algorithms. Compared with 2op, polyphony is simply halved. In other words, a chip can accommodate 9 voices in 2x2op mode.
+
+- Similarly, for 4op algorithms, each voice occupies two channels. However, the channels cannot be occupied arbitrarily. Up to 12 channels may be reserved for 4op, allowing 6 4op voices.
+
+.4op channel reservation
+image::../resources/voices4op.png[400,400]
+
+The example above shows two OPL3 chips with 18 channels each. We have reserved 16 channels for 8 4op voices. Each voice has 2 channels, one colored pink and one colored blue. The remaining 20 channels in gray are available for 2op and 2x2op, allowing for polyphony of 20 or 10 voices respectively.
+
+.Chips
+image::../resources/chip.png[180,180]
+
+Maximum polyphony is controlled by the parameters "Chips" and "4 ops". The "4 ops" parameter is the maximum number of 4op voices, and must be no greater than 6 times the number of chips.
+
+Clicking on image:../resources/emoji_u1f4a1.png[24,24] automatically sets the "4 ops" parameter, based on the number of chips and the currently loaded bank. This value is a suggestion, and may not be optimal.
+
+WARNING: When the selected bank contains 4op programs, and there are no 4op voices reserved, those programs will be inaudible.
+
+=== Arpeggiation
+
+The synthesizer includes intelligent polyphony management that optimizes channel distribution. However, when channels are heavily occupied, and there are insufficient vacant channels to accommodate a new voice, an arpeggiation mechanism is triggered.
+
+The arpeggiation allows a channel to be shared between 2 or more voices, rapidly switching the properties of the channel between the voices using it. This can create an illusion of polyphony, but also an audible distortion, which may sound uncomfortable in some situations.
+
+.Chord played on the "Harpsichord" instrument, without arpeggiation
+audio::../resources/arpeggio-no.opus[]
+
+.Chord played on the "Harpsichord" instrument, with arpeggiation
+audio::../resources/arpeggio-yes.opus[]
+
+Arpeggiation indicates insufficient polyphony. To prevent arpeggiation, the number of chips or 4op voices should be increased.
+
+=== Emulators
+
+One of several YMF262 chip emulators may be selected. Emulators may be switched dynamically.
+
+During the time required for switching emulators, the sound will be muted, and the MIDI stream will be ignored. This will not interfere with real-time operation.
+
+.Emulators
+image::../resources/emulators.png[200,200]
+
+- DOSBox OPL: This emulator is a balanced choice, producing good fidelity sound while using a very reasonable quantity of CPU resources.
+- Nuked OPL: This emulator offers excellent fidelity, aiming to provide nearly cycle-exact emulation. However, CPU usage is significant, so it is only recommended with a small number of chips, or for non-realtime rendering.
```

Comment:
May I add descriptions of the other emulator cores in the project, and the two versions of Nuked?

Reply:
Sure, what we have as v1.7 is the non-vanilla version including optimization patches by Alexander Troosh.
v1.8 has ADSR reimplementation and less efficiency on CPU.